.github/workflows/main.yml

```diff
@@ -0,0 +1,79 @@
+name: CI
+
+on: push
```

Comment:
do we really want to execute workflows on every push to every branch? it makes sense for lint and tests, but not for  deployment job

Reply:
The deployment job has an if statement to filter it only to tags.

Reply:
it would be much safer to create another workflow dedicated to deployment job and not rely on the conditional expression

Reply:
I don't think GitHub Actions supports workflow interdependencies, where one workflow triggers another once completed, based on the branch.

If we make another workflow dedicated to the deployment job, it would be 80% copied code from this.

This is pretty standard practice though.

Reply:
makes sense